### PR TITLE
Add documentation for the external param in volume definitions

### PR DESCRIPTION
**Pull request by shin-**
Also fixed the misdocumented `driver: default`

### CODE REVIEW

File: docs/compose-file.md [hunk 1]
```diff
@@ -576,6 +576,20 @@ documentation for more information. Optional.
...
+become invalid.
+An external volume is considered to exist outside of compose's managed
+application and will not be created or modified. An error will be raised
+if the volume doesn't exist during the compose "up" phase.
+The value for this parameter can either be a boolean or a string. If a
+string is provided, it is considered to be the name of the volume in the
+Docker engine.
+
+    volume01:
+      external: data_d341c8ee
```
Comment (dnephin) [inline docs/compose-file.md]: This isn't quite right, it's actually bool or object with a `name: ...`

**Comment by dnephin** [comment]
cc @londoncalling for docs changes

**Comment by londoncalling** [comment]
LGTM

**Comment by aanand** [comment]
I've incorporated this into #2699, so I'm closing this.

**Pull request closed without merging**